src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlCommand.cs

```diff
@@ -2067,20 +2041,9 @@ private Task InternalExecuteNonQuery(TaskCompletionSource<object> completion, st
 
                     Task task = null;
 
-                    // only send over SQL Batch command if we are not a stored proc and have no parameters and not in batch RPC mode
-                    if (_activeConnection.IsContextConnection)
-                    {
-                        if (statistics != null)
-                        {
-                            statistics.SafeIncrement(ref statistics._unpreparedExecs);
-                        }
-
-                        RunExecuteNonQuerySmi(sendToPipe);
-                    }
-
                     //Always Encrypted generally operates only on parameterized queries. However enclave based Always encrypted also supports unparameterized queries
                     //We skip this block for enclave based always encrypted so that we can make a call to SQL Server to get the encryption information
-                    else if (!ShouldUseEnclaveBasedWorkflow && !_batchRPCMode && (System.Data.CommandType.Text == this.CommandType) && (0 == GetParameterCount(_parameters)))
+                    if (!ShouldUseEnclaveBasedWorkflow && !_batchRPCMode && (System.Data.CommandType.Text == this.CommandType) && (0 == GetParameterCount(_parameters)))
```

Comment:
While you're here, can you remove the redundant parentheses and flip the 0 == condition? :)

Reply:
Yes - done!